### DISCUSSION TITLE
next/image renders incorrect image

### DISCUSSION THREAD

**Discussion by m5p3nc3r** [vercel/next.js#41021] [Help]
### Verify canary release

- [x] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

   Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:23 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.15.0
      npm: 8.5.5
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.3.1
      eslint-config-next: 12.3.0
      react: 18.2.0
      react-dom: 18.2.0

### What browser are you using? (if relevant)

Chrome Version 104.0.5112.101 (Official Build) (arm64)

### How are you deploying your application? (if relevant)

npm run dev

### Describe the Bug

Random incorrect images shown.

### Expected Behavior

The requested image is shown in the browser

### Link to reproduction

https://stackblitz.com/edit/vercel-next-js-c1i3g6

### To Reproduce

I am experimenting with recreating the SOAFEE website using nextjs (https://soafee.io), but am having issues replicating the https://soafee.io/about/members/ page.

I am unable to upload the image resources to the reproduction site, so you will need to add images from:
-  https://gitlab.arm.com/soafee/website/-/tree/main/content/headless_media/images under public/images

If you want to show the (working) number images, please add
- [numbers.zip](https://github.com/vercel/next.js/files/9630050/numbers.zip) to public/numbers

I have randomised the number requests between .png and .jpg to reflect the actual logo loading, but all numbers seem to render correctly and there are no random swapping/duplication of images in the grid.

If I override the loader to use the exact source URL, everything works fine, so it looks like the backend server is serving the wrong image at times.  If the server is restarted and the browser has a forced reload with cleared cache, the errors are different every reload. 

To me, this feels like a race condition on the server side as if I generate a static site from the sources, everything seems to work fine.

**Comment by m5p3nc3r** [comment]
I thought I should add an image of the rendered output of the test code to showcase what I am seeing.

<img width="522" alt="Screenshot 2022-09-23 at 10 14 52" src="https://user-images.githubusercontent.com/1710852/191929399-9396c6f4-cad6-4f56-940b-c5cafca0ff74.png">

You will notice that the AutoCore image has been duplicated.  The second AutoCore image should have been Capgemini, and the CoreAVI image is also incorrect as it should have been Chalmers uni.

If I force an empty cache reload in Chrome, the errors are different on every refresh. 

<img width="513" alt="Screenshot 2022-09-23 at 10 18 44" src="https://user-images.githubusercontent.com/1710852/191929913-e8016660-0d4d-4b87-b00a-77f46e09fcb3.png">

Here you can see the Chamlers logo duplicated when the bottom row should be: AMD, AutoCore, Capgemini, Chalmers.

**Comment by kmsomebody** [comment]
Have you tried it with [sharp](https://www.npmjs.com/package/sharp) installed, as [documented here](https://nextjs.org/docs/api-reference/next/image#built-in-loaders)?
I had the same issue before using sharp.

**Comment by m5p3nc3r** [comment]
Hey @DarthSomebody, I have changed to using sharp, and it seems to have solved my problem.

The issue is, I don't understand why the default behaviour is Ok to be incorrect?  Does the fact that the issue has now been changed to a discussion mean the root cause of the problem is not going to be resolved?
